### PR TITLE
Expand pools to higher token counts

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Now that the Vault supports 8 tokens, we can move the Weighted Pool max up to 8, and restore v2's Stable Pool max token count of 5 (limited by StableMath; higher token counts are possible, but haven't been verified to work to the extent that 5-token pools were).

This removes a limitation on v3 that turns out to have been unnecessary, and restores parity with v2. Should make some partners happy!

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/pool-weighted/contracts/WeightedPool.sol [hunk 1]
```diff
@@ -166,6 +178,26 @@ contract WeightedPool is IWeightedPool, BalancerPoolToken, PoolInfo, Version {
...
+        if (numTokens > 6) {
+            normalizedWeights[6] = _normalizedWeight6;
+        } else {
+            return normalizedWeights;
+        }
+        if (numTokens > 7) {
+            normalizedWeights[7] = _normalizedWeight7;
+        } else {
+            return normalizedWeights;
+        }
```
Comment (jubeira) [inline pkg/pool-weighted/contracts/WeightedPool.sol:200]: This one's not needed, but also not wrong.

File: pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    function testInitialize() public view {
+        (, , uint256[] memory balances, ) = vault.getPoolTokenInfo(address(pool));
+
+        for (uint256 i = 0; i < numTokens; ++i) {
+            // Tokens are transferred from lp
+            assertEq(
+                TOKEN_AMOUNT - bigPoolTokens[i].balanceOf(lp),
+                TOKEN_AMOUNT,
+                string.concat("LP: Wrong token balance for ", Strings.toString(i))
+            );
```
Comment (jubeira) [inline pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol:165]: Maybe just say `balanceOf(lp)` equals 0?

File: pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+            // amountsOut are correct
+            assertApproxEqAbs(
+                amountsOut[i],
+                TOKEN_AMOUNT_IN,
+                TOKEN_DELTA,
+                string.concat("Pool: Wrong amountOut for ", Strings.toString(i))
+            );
+        }
+
+        // should mint correct amount of BPT tokens
```
Comment (jubeira) [inline pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol:283]:
```suggestion
        // should burn correct amount of BPT tokens
```

**Review by jubeira — APPROVED** [review]
LGTM, thanks @EndymionJkb ! Great job.

File: pkg/pool-weighted/contracts/WeightedPool.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/pool-weighted/contracts/WeightedPool.sol:200]: Made it look more like v2, and removed the last one.

File: pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol (hunk 3 above)
Comment (EndymionJkb) [reply pkg/pool-weighted/test/foundry/BigWeightedPool.t.sol:283]: Good catch. Thanks!

**Pull request merged**